types_test.go

```diff
@@ -0,0 +1,104 @@
+// This code has been cherry-picked from https://github.com/go-openapi/strfmt
+
+package strfmt
+
+import (
+	"bytes"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	p, _ = time.Parse(time.RFC3339Nano, "2011-08-18T19:03:37.000000000+01:00")
+
+	testCases = []struct {
+		in     []byte    // externally sourced data -- to be unmarshalled
+		time   time.Time // its representation in time.Time
+		str    string    // its marshalled representation
+		utcStr string    // the marshaled representation as utc
+	}{
+		{[]byte("2014-12-15 08:00:00"), time.Date(2014, 12, 15, 8, 0, 0, 0, time.UTC), "2014-12-15T08:00:00.000Z", "2014-12-15T08:00:00.000Z"},
+		{[]byte("2014-12-15T08:00:00"), time.Date(2014, 12, 15, 8, 0, 0, 0, time.UTC), "2014-12-15T08:00:00.000Z", "2014-12-15T08:00:00.000Z"},
+		{[]byte("2014-12-15T08:00"), time.Date(2014, 12, 15, 8, 0, 0, 0, time.UTC), "2014-12-15T08:00:00.000Z", "2014-12-15T08:00:00.000Z"},
+		{[]byte("2014-12-15T08:00Z"), time.Date(2014, 12, 15, 8, 0, 0, 0, time.UTC), "2014-12-15T08:00:00.000Z", "2014-12-15T08:00:00.000Z"},
+		{[]byte("2018-01-28T23:54Z"), time.Date(2018, 01, 28, 23, 54, 0, 0, time.UTC), "2018-01-28T23:54:00.000Z", "2018-01-28T23:54:00.000Z"},
+		{[]byte("2014-12-15T08:00:00.000Z"), time.Date(2014, 12, 15, 8, 0, 0, 0, time.UTC), "2014-12-15T08:00:00.000Z", "2014-12-15T08:00:00.000Z"},
+		{[]byte("2011-08-18T19:03:37.123000000+01:00"), time.Date(2011, 8, 18, 19, 3, 37, 123*1e6, p.Location()), "2011-08-18T19:03:37.123+01:00", "2011-08-18T18:03:37.123Z"},
+		{[]byte("2011-08-18T19:03:37.123000+0100"), time.Date(2011, 8, 18, 19, 3, 37, 123*1e6, p.Location()), "2011-08-18T19:03:37.123+01:00", "2011-08-18T18:03:37.123Z"},
+		{[]byte("2011-08-18T19:03:37.123+0100"), time.Date(2011, 8, 18, 19, 3, 37, 123*1e6, p.Location()), "2011-08-18T19:03:37.123+01:00", "2011-08-18T18:03:37.123Z"},
+		{[]byte("2014-12-15T19:30:20Z"), time.Date(2014, 12, 15, 19, 30, 20, 0, time.UTC), "2014-12-15T19:30:20.000Z", "2014-12-15T19:30:20.000Z"},
+		{[]byte("0001-01-01T00:00:00Z"), time.Time{}.UTC(), "0001-01-01T00:00:00.000Z", "0001-01-01T00:00:00.000Z"},
+		{[]byte(""), time.Unix(0, 0).UTC(), "1970-01-01T00:00:00.000Z", "1970-01-01T00:00:00.000Z"},
+		{[]byte(nil), time.Unix(0, 0).UTC(), "1970-01-01T00:00:00.000Z", "1970-01-01T00:00:00.000Z"},
+	}
+)
+
+func TestNewDateTime(t *testing.T) {
+	assert.EqualValues(t, time.Unix(0, 0).UTC(), NewDateTime())
+}
+
+func TestParseDateTime_errorCases(t *testing.T) {
+	_, err := ParseDateTime("yada")
+	assert.Error(t, err)
+}
+
+func TestDateTime_UnmarshalJSON(t *testing.T) {
+	for caseNum, example := range testCases {
+		t.Logf("Case #%d", caseNum)
+		pp := NewDateTime()
+		err := pp.UnmarshalJSON(esc(example.in))
+		assert.NoError(t, err)
+		assert.EqualValues(t, example.time, pp)
```

Comment:
I think we can improve the test run output readability by using the [Run](https://pkg.go.dev/testing#T.Run) function. As we have started to do in other [repositories](https://github.com/kubewarden/kubewarden-controller/blob/d7aa076ce67ecac1ab1bcda0ec4407ff85a16575/internal/pkg/admission/reconciler_test.go#L77).

Reply:
This comment applies for the other tests that you have here. ;)

Reply:
I just saw this comment:

```
// This code has been cherry-picked from https://github.com/go-openapi/strfmt
```

If we plan to keep this file and apply some future patches from the upstream repo. This change request may cause some conflicts...

Reply:
Yes, the idea was not to refactor this to be as close as possible to upstream.

Reply:
yeah, let's stay as close to upstream as possible